docs: add docs for issue explorer (#760)
infracost/docs: docs/infracost_cloud/issue_explorer.md

## Key Benefits
- **Centralized visibility**: Track all infrastructure issues across multiple repositories and teams in one place
- **Prioritized actions**: Focus on high-impact issues with the greatest cost-saving potential
- **Team accountability**: Assign and track issue resolution across engineering teams
- **Compliance tracking**: Monitor adherence to FinOps policies and governance standards
- **Historical trends**: Understand how your organization's cloud governance improves over time

## Before You Begin

Here's what you need to get started:

* ✅ Infracost Cloud account with appropriate permissions
* ✅ At least one repository integrated (GitHub, GitLab, or Azure Repos)

## Getting Started

You can access Issue Explorer by selecting `Visibility > Issue explorer` from the main 
navigation menu:

![Where to find Issue Explorer in the main navigation](/img/issue-explorer/issue-explorer-nav.png)

### Switching Chart Type

> [PLACEHOLDER: Add dashboard screenshot]

Reviewer: ```suggestion
```

Click either chart icon to instantly switch views—your filters and date ranges remain unchanged, letting you explore the same data from different perspectives.

### Metrics & Grouping

#### Available Metrics

![Selecting the metric to aggregate](/img/issue-explorer/issue-explorer-metrics.png)

Customize your analysis by selecting which metrics to track and how to organize them. This flexibility enables you to answer specific questions and uncover actionable insights.

Choose which issue metric to sum in your reports:

| Attribute | Shows | Use when |
| --------- | ----- | -------- |
| Open Issues | Current unresolved issues requiring attention | Identifying where to focus remediation efforts, understanding current technical debt, or planning campaigns |
| Fixed Issues | Issues that have been successfully resolved | Demonstrating team progress, celebrating wins, or identifying which teams/policies are seeing the most improvement |

> | Prevented Issues | Issues caught and blocked before reaching production | Measuring shift-left success, validating CI/CD policies, or showing ROI of prevention tooling |

Reviewer: ```suggestion
| Prevented Issues | Issues caught, and either prevented before reaching production or fixed by follow-up changes | Measuring shift-left success, validating CI/CD policies, or showing ROI of prevention tooling |
```

| New Issues | Recently introduced issues within your selected timeframe | Identifying regression trends, spotting problematic deployments, or understanding if governance is improving |
| Dismissed Issues | Issues marked as false positives or accepted risks | Auditing policy effectiveness, understanding exception patterns, or refining policy rules |

#### Grouping Options

![Selecting the attribute to group by](/img/issue-explorer/issue-explorer-groupby.png)

Organize your selected metric by different attributes to reveal patterns:

| Group by | Reveals | Example insights |
| -------- | ------- | ------- |

> | Policy | Which FinOps policies are most/least violated | "Unattached disk" violations are increasing while "preferred instance" violations are decreasing

Reviewer: ```suggestion
| Policy | Which FinOps policies are most/least violated | "gp2 volume" violations are increasing while "preferred instance" violations are decreasing
```

| Repository | Which codebases contribute most to your issues | The payments-service repo has fixed 90% of issues while the legacy-api repo shows no improvement |
| Repository Org | Patterns across GitHub/GitLab organizations | The platform-team org prevents 5x more issues than other orgs due to better CI/CD checks |

> | `CustomAttribute` | Patterns specific to your organization | _(Options include: Business unit, team, application, service, cost center, or any configured attribute)_ <br/><br/>The Data & Analytics business unit has the highest prevented-to-fixed ratio, showing strong shift-left adoption |

Reviewer: ```suggestion
| `CustomAttribute` | Patterns specific to your organization | _(Options can include: Business unit, team, application, service, cost center, or any configured attribute)_ <br/><br/>The Data & Analytics business unit has the highest prevented-to-fixed ratio, showing strong shift-left adoption |
```

| _"Are we getting better or worse?"_ | New Issues | Policy | 
| _"Which repos need the most help?"_ | Open Issues | Repository | 
| _"What exceptions are we accepting?"_ | Dismissed Issues | Policy Category |

### Filtering

Focus your analysis on exactly the data you need using Issue Explorer's comprehensive filtering 
system. Combine date ranges with attribute filters to create precise views of your 
infrastructure issues.

#### Date Range Filtering 

![Filtering by date range](/img/issue-explorer/issue-explorer-daterange.png)

Set custom time periods to analyze trends, or focus on recent changes:

* Preset ranges: Last 7, 30, 90, or 365 days
* Custom ranges: Select specific start and end dates

#### Available Attribute Filters

##### Campaign and Policy Filters

* **Campaign:** View issues associated with specific campaigns
* **Policy Name:** Filter by individual FinOps policy
* **Policy Category:** Group by policy types (tagging, cost optimization, security)

##### Development Filters

* **Repo Name:** Focus on specific repositories
* **Repo Org:** Filter by GitHub/GitLab organization
* **Commit Author:** Track issues by who introduced them

> * **PR Comment:** Find issues flagged in pull request reviews

Reviewer: ```suggestion
* **PR Comment:** Find issues flagged (i.e. enabled) in pull request reviews
```
